utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
 const puppeteer = require('puppeteer'); // v13.0.0 or later
 
-const getWaifu = async () => {
-  const waifu = {
-    name: '',
-    imageUrl: '',
-    bioUrl: '',
-    extract: '',
-    appearsIn: '',
+const nameSelector = 'h1';
+const imageSelector = '#widget-waifu-of-the-day > a > img';
+const popularitySelector = '#popularity-rank';
+const ageSelector = '#age';
+const husbandoSelector = '#waifu-classification';
+const appearsSelector = '#waifu-core-information > div > div > a';
+
+const waifuURL = "https://mywaifulist.moe/"
+const randomUrl = "https://mywaifulist.moe/random"
+
+const getWaifu = async (retry = false) => {
+  const [browser, page] = await intializeScrapper()
+  await page.setViewport({ "width": 1920, "height": 1080 })
+
+  const url = retry ? randomUrl : waifuURL
```

Comment:
Seems that the `retry` argument is actually being used to select the URL (random or WotD)... please change it to a more explicit name, i.e. `isFromRandom` | `isWOTD`.

Another (and probably better) approach would be to simply pass in the URL to use in `getWaifu`.